libpod/network/netconflist.go

```diff
@@ -267,3 +254,82 @@ func IfPassesFilter(netconf *libcni.NetworkConfigList, filters map[string][]stri
 	}
 	return result, nil
 }
+
+// ComputeUntilTimestamp extracts unitil timestamp from filters
+func ComputeUntilTimestamp(filter string, filterValues []string) (*time.Time, error) {
+	if len(filterValues) != 1 {
+		return nil, errors.Errorf("specify exactly one timestamp for %s", filter)
+	}
+	ts, err := timetype.GetTimestamp(filterValues[0], time.Now())
+	if err != nil {
+		return nil, err
+	}
+	seconds, nanoseconds, err := timetype.ParseTimestamps(ts, 0)
+	if err != nil {
+		return nil, err
+	}
+	until := time.Unix(seconds, nanoseconds)
+	return &until, nil
+}
+
+// IfPassesPruneFilter filters NetworkListReport and returns true if the prune filter match the given config
+func IfPassesPruneFilter(config *config.Config, netconf *libcni.NetworkConfigList, f map[string][]string) (bool, error) {
+	for key, filterValues := range f {
+		switch strings.ToLower(key) {
+		case "label":
+			return matchPruneLabelFilters(netconf, filterValues), nil
+		case "until":
+			until, err := ComputeUntilTimestamp(key, filterValues)
+			if err != nil {
+				return false, err
+			}
+			created, err := getCreatedTimestamp(config, netconf)
+			if err != nil {
+				return false, err
+			}
+			if created.Before(*until) {
+				return true, nil
+			}
+		default:
+			return false, errors.Errorf("invalid filter %q", key)
+		}
+	}
+	return false, nil
+}
+
+func matchPruneLabelFilters(netconf *libcni.NetworkConfigList, filterValues []string) bool {
+	labels := GetNetworkLabels(netconf)
+	result := true
+outer:
+	for _, filterValue := range filterValues {
+		filterArray := strings.SplitN(filterValue, "=", 2)
+		filterKey := filterArray[0]
+		if len(filterArray) > 1 {
+			filterValue = filterArray[1]
+		} else {
+			filterValue = ""
+		}
+		for labelKey, labelValue := range labels {
+			if labelKey == filterKey && (filterValue == "" || labelValue == filterValue) {
+				result = true
+				continue outer
+			}
+		}
+		result = false
+	}
+	return result
+}
+
+func getCreatedTimestamp(config *config.Config, netconf *libcni.NetworkConfigList) (*time.Time, error) {
+	networkConfigPath, err := GetCNIConfigPathByNameOrID(config, netconf.Name)
+	if err != nil {
+		return nil, err
+	}
+	f, err := os.Stat(networkConfigPath)
+	if err != nil {
+		return nil, err
+	}
+	stat := f.Sys().(*syscall.Stat_t)
```

Comment:
@containers/podman-maintainers Windows and OS X cross builds are failing because of syscall.Stat_t. As far as I know, there is no way to get timestamp created from data structures as in the case of the containers. What would be your recommended way to pass cross builds? Making arch-specific code here?

Reply:
I will open a PR to fix this. We should not use `libpod/network` client side.

Reply:
@Luap99 Ok, thanks. Let me know and then I will do a rebase.

Reply:
You can try with #9716

Reply:
done